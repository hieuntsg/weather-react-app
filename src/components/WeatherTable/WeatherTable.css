.paddedBox{
  padding: 12px;
}

.clickableBox:hover {
  cursor: pointer;
  background: rgba(100,100,100,0.1)
}

.noMargin {
  margin:0;
  padding:0;
}