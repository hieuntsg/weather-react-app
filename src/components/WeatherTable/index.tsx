import * as React from 'react';

import { Weather } from '../../types/weather';
import moment from "moment";
import './WeatherTable.css';

type Props = {
  data: Weather
};

const WeatherTable: React.FC<Props> = ({ data }) => {
  const month_map: any = {
    1: 'Jan',
    2: 'Feb',
    3: 'Mar',
    4: 'Apr',
    5: 'May',
    6: 'Jun',
    7: 'Jul',
    8: 'Aug',
    9: 'Sep',
    10: 'Oct',
    11: 'Nov',
    12: 'Dec'
  };
  if (Object.keys(data.dailyData).length === 0) return (null);

  return <React.Fragment>
    <hr />
    <nav className="level">
      {Object.keys(data.dailyData).map((date, i) => (
        <div key={`daily${date}`} className="level-item has-text-centered clickableBox paddedBox">
          <div>
            <p>
              {parseInt(date.split("-")[2])}&nbsp;{month_map[parseInt(date.split("-")[1])]}&nbsp;{parseInt(date.split("-")[0])}
              <br />
              <p className="is-size-4">{moment(date).utc().format("dddd")}</p>
            </p>
            <img src={`https://openweathermap.org/img/wn/${data.dailyData[date]["icon"]}@2x.png`} width="50px" height="50px" alt="icon" />
            <br />

            {parseFloat(data.dailyData[date]["temp_min"]).toFixed(1)} - {parseFloat(data.dailyData[date]["temp_max"]).toFixed(1)}Â°C<br />

            {(data.dailyData[date]["weather"])}<br />

              Wind: {(data.dailyData[date]["wind_max"])}m/s

            </div>
        </div>
      ))}
    </nav>
    <hr />
  </React.Fragment>
};

export default WeatherTable;
