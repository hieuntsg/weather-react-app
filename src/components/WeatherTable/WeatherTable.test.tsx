import React from "react";
import { shallow } from "enzyme";

import WeatherTable from "./";

describe("<WeatherTable />", () => {
  it("renders without crashing", () => {
    const ICON_URL = "https://openweathermap.org/img/w/";
    const data = {
      city: "Madrid ES",
      list: [
        {
          day: "Tuesday",
          time: "3:00PM",
          temp: "13",
          icon: `${ICON_URL}03d.png`,
          dt: 1556031600,
        },
        {
          day: "Tuesday",
          time: "6:00PM",
          temp: "10",
          icon: `${ICON_URL}03d.png`,
          dt: 1556042400,
        },
        {
          day: "Tuesday",
          time: "9:00PM",
          temp: "7",
          icon: `${ICON_URL}03d.png`,
          dt: 1556053200,
        },
        {
          day: "Wednesday",
          time: "12:00AM",
          temp: "6",
          icon: `${ICON_URL}03d.png`,
          dt: 1556064000,
        },
      ],
      dailyData: {
        "2022-11-11": {
          temp_max: 14.3,
          temp_min: 11.27,
          weather_id: 803,
          weather: "broken clouds",
          wind_max: 3.38,
          icon: "04n",
        },
        "2022-11-12": {
          temp_max: 16.07,
          temp_min: 10.39,
          weather_id: 804,
          weather: "overcast clouds",
          wind_max: 2.85,
          icon: "04n",
        },
        "2022-11-13": {
          temp_max: 15.23,
          temp_min: 11.3,
          weather_id: 804,
          weather: "overcast clouds",
          wind_max: 2.43,
          icon: "04n",
        },
        "2022-11-14": {
          temp_max: 13.35,
          temp_min: 10.54,
          weather_id: 500,
          weather: "light rain",
          wind_max: 1.88,
          icon: "10d",
        },
        "2022-11-15": {
          temp_max: 13.44,
          temp_min: 9.72,
          weather_id: 501,
          weather: "moderate rain",
          wind_max: 6.85,
          icon: "10n",
        },
        "2022-11-16": {
          temp_max: 11.05,
          temp_min: 8.81,
          weather_id: 500,
          weather: "light rain",
          wind_max: 5.63,
          icon: "10n",
        },
      },
    };
    const wrapper = shallow(<WeatherTable data={data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
