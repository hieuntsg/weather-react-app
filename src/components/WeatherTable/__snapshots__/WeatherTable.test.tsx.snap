// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherTable /> renders without crashing 1`] = `
<Fragment>
  <hr />
  <nav
    className="level"
  >
    <div
      className="level-item has-text-centered clickableBox paddedBox"
      key="daily2022-11-11"
    >
      <div>
        <p>
          11
           
          Nov
           
          2022
          <br />
          <p
            className="is-size-4"
          >
            Thursday
          </p>
        </p>
        <img
          alt="icon"
          height="50px"
          src="https://openweathermap.org/img/wn/04n@2x.png"
          width="50px"
        />
        <br />
        11.3
         - 
        14.3
        °C
        <br />
        broken clouds
        <br />
        Wind: 
        3.38
        m/s
      </div>
    </div>
    <div
      className="level-item has-text-centered clickableBox paddedBox"
      key="daily2022-11-12"
    >
      <div>
        <p>
          12
           
          Nov
           
          2022
          <br />
          <p
            className="is-size-4"
          >
            Friday
          </p>
        </p>
        <img
          alt="icon"
          height="50px"
          src="https://openweathermap.org/img/wn/04n@2x.png"
          width="50px"
        />
        <br />
        10.4
         - 
        16.1
        °C
        <br />
        overcast clouds
        <br />
        Wind: 
        2.85
        m/s
      </div>
    </div>
    <div
      className="level-item has-text-centered clickableBox paddedBox"
      key="daily2022-11-13"
    >
      <div>
        <p>
          13
           
          Nov
           
          2022
          <br />
          <p
            className="is-size-4"
          >
            Saturday
          </p>
        </p>
        <img
          alt="icon"
          height="50px"
          src="https://openweathermap.org/img/wn/04n@2x.png"
          width="50px"
        />
        <br />
        11.3
         - 
        15.2
        °C
        <br />
        overcast clouds
        <br />
        Wind: 
        2.43
        m/s
      </div>
    </div>
    <div
      className="level-item has-text-centered clickableBox paddedBox"
      key="daily2022-11-14"
    >
      <div>
        <p>
          14
           
          Nov
           
          2022
          <br />
          <p
            className="is-size-4"
          >
            Sunday
          </p>
        </p>
        <img
          alt="icon"
          height="50px"
          src="https://openweathermap.org/img/wn/10d@2x.png"
          width="50px"
        />
        <br />
        10.5
         - 
        13.3
        °C
        <br />
        light rain
        <br />
        Wind: 
        1.88
        m/s
      </div>
    </div>
    <div
      className="level-item has-text-centered clickableBox paddedBox"
      key="daily2022-11-15"
    >
      <div>
        <p>
          15
           
          Nov
           
          2022
          <br />
          <p
            className="is-size-4"
          >
            Monday
          </p>
        </p>
        <img
          alt="icon"
          height="50px"
          src="https://openweathermap.org/img/wn/10n@2x.png"
          width="50px"
        />
        <br />
        9.7
         - 
        13.4
        °C
        <br />
        moderate rain
        <br />
        Wind: 
        6.85
        m/s
      </div>
    </div>
    <div
      className="level-item has-text-centered clickableBox paddedBox"
      key="daily2022-11-16"
    >
      <div>
        <p>
          16
           
          Nov
           
          2022
          <br />
          <p
            className="is-size-4"
          >
            Tuesday
          </p>
        </p>
        <img
          alt="icon"
          height="50px"
          src="https://openweathermap.org/img/wn/10n@2x.png"
          width="50px"
        />
        <br />
        8.8
         - 
        11.1
        °C
        <br />
        light rain
        <br />
        Wind: 
        5.63
        m/s
      </div>
    </div>
  </nav>
  <hr />
</Fragment>
`;
